sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/BeginInvokePairedWithEndInvoke.cs

```diff
@@ -67,7 +67,7 @@ public sealed class BeginInvokePairedWithEndInvoke : SonarDiagnosticAnalyzer
                 {
                     var invocation = (InvocationExpressionSyntax)c.Node;
                     const string BeginInvoke = "BeginInvoke";
-                    if (invocation.ToStringContains(BeginInvoke) &&
+                    if (invocation.Expression.ToStringContains(BeginInvoke) &&
```

Comment:
this is to avoid what you noticed about checking the params as well.

Reply:
Sneaky technical debt killed with fire!